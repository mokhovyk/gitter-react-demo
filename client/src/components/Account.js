import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class Account extends Component {
  render() {
    const { isAuthorized, userId, userName, displayName, avatarUrl } = this.props;

    return (
      <ul className="list-group">
        <li className="list-group-item"><h2>My account</h2></li>
        <li className="list-group-item"><img src={avatarUrl} alt="avatar" /></li>
        <li className="list-group-item">userId: {userId}</li>
        <li className="list-group-item">userName: {userName}</li>
        <li className="list-group-item">displayName: {displayName}</li>
      </ul>
    );
  }
}

const mapStateToProps = ({ user }) => ({
  ...user,
});

export default connect(mapStateToProps)(Account);
