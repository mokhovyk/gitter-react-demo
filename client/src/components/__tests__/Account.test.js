import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Account, { AccountComponent} from '../Account';

let props;
let component;

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Account', () => {
  beforeEach(() => {
    props = {
      userId: 'userId123',
      userName: 'johndoe',
      displayName: 'John Doe',
      avatarUrl: 'avatar.png',
    };
    component = shallow(
      <AccountComponent {...props} />,
    );
  });

  test('renders correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test('renders connected container', () => {
    const initialState = {
      user: {
        userId: 'userId123',
        userName: 'johndoe',
        displayName: 'John Doe',
        avatarUrl: 'avatar.png',
      },
    };
    const store = mockStore(initialState);
    const container = shallow(
      <AccountComponent />,
      { context: { store } },
    );

    expect(toJson(container)).toMatchSnapshot();
  });
});
