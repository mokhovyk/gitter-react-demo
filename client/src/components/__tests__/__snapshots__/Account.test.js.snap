// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account renders connected container 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <h2>
      My account
    </h2>
  </li>
  <li
    className="list-group-item"
  >
    <img
      alt="avatar"
    />
  </li>
  <li
    className="list-group-item"
  >
    userId: 
  </li>
  <li
    className="list-group-item"
  >
    userName: 
  </li>
  <li
    className="list-group-item"
  >
    displayName: 
  </li>
</ul>
`;

exports[`Account renders correctly 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <h2>
      My account
    </h2>
  </li>
  <li
    className="list-group-item"
  >
    <img
      alt="avatar"
      src="avatar.png"
    />
  </li>
  <li
    className="list-group-item"
  >
    userId: 
    userId123
  </li>
  <li
    className="list-group-item"
  >
    userName: 
    johndoe
  </li>
  <li
    className="list-group-item"
  >
    displayName: 
    John Doe
  </li>
</ul>
`;
